themes/classic/templates/_partials/microdata-jsonld.tpl

```diff
@@ -0,0 +1,201 @@
+      "name": "{$shop.name}"
+    },
+    "name": "{$page.meta.title}",
+    "url":  "{$urls.current_url}"
+  }
+</script>
+
+{if $page.page_name =='index'}
+  <script type="application/ld+json">
+    {
+      "@context": "https://schema.org",
+      "@type": "WebSite",
+      "url" : "{$urls.pages.index}",
+      "image": {
+        "@type": "ImageObject",
+        "url":"{$urls.shop_domain_url}{$shop.logo}"
+      },
+      "potentialAction": {
+        "@type": "SearchAction",
+        "target": "{'--search_term_string--'|str_replace:'{search_term_string}':$link->getPageLink('search',true,null,['search_query'=>'--search_term_string--'])}",
+        "query-input": "required name=search_term_string"
+      }
+    }
+  </script>
+{/if}
+
+{if isset($product) && $page.page_name == 'product'}
+  <script type="application/ld+json">
+    {
+      "@context": "https://schema.org/",
+      "@type": "Product",
+      "name": "{$product.name}",
+      "description": "{$page.meta.description|regex_replace:"/[\r\n]/" : " "}",
+      "category": "{$product.category_name}",
+      {if isset($product.cover)}"image" :"{$product.cover.bySize.home_default.url}",{/if}
+      {if $product.reference}"sku": "{$product.reference}",{else}"sku": "{$product.id}",{/if}
+      "mpn": {if $product.reference}"{$product.reference}"{else}"{$product.id}"{/if},
+      {if $product_manufacturer->name OR $shop.name}"brand": {
+        "@type": "Thing",
+        "name": "{if $product_manufacturer->name}{$product_manufacturer->name|escape:'html':'UTF-8'}{else}{$shop.name}{/if}"
+      },{/if}
+      {if isset($nbComments) && $nbComments && $ratings.avg}"aggregateRating": {
+        "@type": "AggregateRating",
+        "ratingValue": "{$ratings.avg|round:1|escape:'html':'UTF-8'}",
+        "reviewCount": "{$nbComments|escape:'html':'UTF-8'}"
+      },{/if}
+      {if isset($product.weight) && ($product.weight != 0)}
+        "weight": {
+          "@context": "https://schema.org",
+          "@type": "QuantitativeValue",
+          "value": "{$product.weight}",
+          "unitCode": "{$product.weight_unit}"
+        },{/if}
+        {if $product.show_price}
+          {if $product.id_product_attribute == 0}
+            "offers": {
+              "@type": "Offer",
+              "priceCurrency": "{$currency.iso_code}",
+              "name": "{$product.name|strip_tags:false}",
+              "price": "{$product.price_amount}",
+              "url": "{$product.url}",
+              "priceValidUntil": "{($smarty.now + (int) (60*60*24*15))|date_format:"%Y-%m-%d"}",
+              {if $product.images|count > 0}
+                "image": {strip}[
+                  {foreach from=$product.images item=p_img name="p_img_list"}
+                    "{$p_img.large.url}"{if not $smarty.foreach.p_img_list.last},{/if}
+                  {/foreach}
+                ]{/strip},
+              {/if}
+              "mpn": {if $product.reference}"{$product.reference}"{else}"{$product.id}"{/if},
```

Comment:
`{if $product.mpn}"mpn": "{$product.mpn}",{/if}`

Reply:
Good prop, I added an additionnal control for product lvl declaration and for offer when no combination => npm first, if not available reference, if not available product_id to avoid Google Rich Text unecessary warning for a manufacturer id.

Reply:
Normally, when you will test with Google rich text. You always will have proper definition for product and offer. (only 2 remaining warnings for reviews and ratings)